### ISSUE TITLE
An issue with rendering raster layers on Android

### ISSUE THREAD

**Issue by maxtracking** [maplibre/maplibre-gl-native#161]
Hello,

I'm not sure why but with the latest updates from 9.5.0 to 9.5.1 my raster layers stopped working. I tracked it down to [this commit](https://github.com/maplibre/maplibre-gl-native/commit/fe041b80caca953d5ffbea19b5d2a36484d0286a), that is, the previous one works OK, but above one doesn't.

Environment: Android 11, API 30, RasterLayer

Any suggestions are appreciated. 

Thanks
Alex

**Comment by maxtracking** [comment]
Maybe @ntadej could comment on above as it's his PR. I cannot be entirely sure but it looks that way

**Comment by ntadej** [comment]
Stopped working in what sense?

**Comment by maxtracking** [comment]
Hi Tadej,

I've tried just the demo app - AnimatedImageSourceActivity. I don't see the animated 'radar' layer there unless I've done something silly. We have our own small changes over MapLibre, so I noticed that after I merged changes from 9.5.1 our raster-based layers stopped appearing or appeared at random, ie at some zoom level only. I then went on to checkout commits one by one and finally stumbled upon that particular one (see in the original post). I decided to double check it on the demo with the same result. 
Could you confirm that after the following animated image layer works ok:
1. Pull master branch from MapLibre
2. Update submodules
3. use armv8 & x86_64 abis for mapbox.abis
4. do whatever's necessary to get styles working, ie either MapBox tokens/keys or MapTiler or whatever. We used our own server so style pointer to a custom url with style json. I've also tried with default MapBox styles.

Thanks a lot,
Alex

**Comment by ntadej** [comment]
Thanks for the reproducer. I'll try to have a look ASAP but probably I won't have time before weekend.

Note that that commit should only fix warnings, but I guess something slipped our attention. In the meantime you can try going through changes and finding suspicious lines.

**Comment by maxtracking** [comment]
Yeah, that's what I'm doing at the moment :) At least I've confirmed it's not just me, my colleague saw the same as I.

**Comment by maxtracking** [comment]
I think I found the issue - it's in `src/mbgl/util/tile_coordinate.hpp` where there is a `static_cast<int64_t>`. If I add brackets around the whole result of multiplication then it works:
```
static GeometryCoordinate toGeometryCoordinate(const UnwrappedTileID& tileID, const TileCoordinatePoint& point) {
        const double scale = std::pow(2.0, tileID.canonical.z);
        auto zoomed = TileCoordinate { point, 0 }.zoomTo(tileID.canonical.z);
        return {
            int16_t(util::clamp<int64_t>(static_cast<int64_t>((zoomed.p.x - tileID.canonical.x - tileID.wrap * scale) * util::EXTENT),
                        std::numeric_limits<int16_t>::min(),
                        std::numeric_limits<int16_t>::max())),
            int16_t(util::clamp<int64_t>(static_cast<int64_t>((zoomed.p.y - tileID.canonical.y) * util::EXTENT),
                        std::numeric_limits<int16_t>::min(),
                        std::numeric_limits<int16_t>::max()))
        };
    }
```
And it happened to be used in `render_image_source.cpp` among few other places so `RasterLayer` would be affected.

**Comment by ntadej** [comment]
OK, let's revert this change for now. I can open a MR in the next days.

**Comment by maxtracking** [comment]
Thanks. I'd suggest to just add 2 enclosing brackets to each pair as shown above rather than reverting that change completely. After all, the intention was to fix the warning so you just wanted to cast the result of the multiplication, not the first item.
